### PR TITLE
Update types for 64 bit and GL

### PR DESCRIPTION
**Pull request by jopadan**
CHANGES:
 - replace all occurrences of long types with unsigned 32-bit integer types
- use predefined types of glcorearb.h if GLCOREARB is defined
- rename existing types to style `[f/u/i]bits` in `bftypes.h`

TODO:
- port assembler code
- #44
- #27
- #20
- #3

### CODE REVIEW

**Comment by mefistotelis** [comment]
What is your rationale for basing all types on OpenGL definitions?
It is unusual to base a whole game on definitions from a specific rendering engine.
(which the game doesn't even use,  currently)

The types change is large enough that it shouldn't be mixed with anything else. Some of the `gitignore` changes seem to secure for a situation where someone does not compile the game in compliance with instructions, so these need a separate discussion.

It is worth mentioning that the type names - while identical to ones defined by Watcom compiler - are really coming from Bullfrog - we have some old headers shared by former employees, and even at the pint when Amiga was still supported as a target, the header did contained:

```
/* BASIC TYPES */
typedef	long					SLONG;
typedef	unsigned	long		ULONG;
typedef	short		int			SWORD;
typedef	unsigned	short	int	UWORD;
typedef	char					SBYTE;
typedef	unsigned 	char		UBYTE;
/* POINTER TYPES */
typedef	SBYTE					*	PSBYTE;
typedef	UBYTE					*	PUBYTE;
typedef	SWORD					*	PSWORD;
typedef	UWORD					*	PUWORD;
typedef	SLONG					*	PSLONG;
typedef	ULONG					*	PULONG;
```

That being said, I do recognize the problem with `long` types vs 64-bit arch.
I also did switched the name 'word' to 'short' in my definitions..

Anyway, we probably need a discussion before a patch.

**Comment by jopadan** [comment]
> What is your rationale for basing all types on OpenGL definitions?
OpenGL types are the most portable platform independent based on standard C/C++ ones.

> Some of the `gitignore` changes seem to secure for a situation where someone does not compile the game in compliance with instructions, so these need a separate discussion.
No success in ignoring m4 directories is the reason for the additional m4 ignores

File: bflibrary/src/general/rnc_1fm.c [hunk 1]
```diff
@@ -437,14 +437,15 @@ long rnc_unpack(void *packed, void *unpacked, unsigned int flags
     return ret_len;
 }
 
-long UnpackM1(unsigned char *buffer, ulong bufsize)
+i32 UnpackM1(unsigned char *buffer, u32 bufsize)
 {
+    fprintf(stdout, "%u\n", bufsize);
```
Comment (jopadan) [inline bflibrary/src/general/rnc_1fm.c:442]: Adding debug output seems to circumvent the segfault race condition/threading/timing caused.

File: bflibrary/src/general/rnc_1fm.c [hunk 2]
```diff
@@ -437,14 +437,15 @@ long rnc_unpack(void *packed, void *unpacked, unsigned int flags
     return ret_len;
 }
 
-long UnpackM1(unsigned char *buffer, ulong bufsize)
+i32 UnpackM1(unsigned char *buffer, u32 bufsize)
 {
+    LOGWARN("UnpackM1: (expected size %zu)", bufsize);
```
Comment (jopadan) [inline bflibrary/src/general/rnc_1fm.c:442]: Adding debug output seems to circumvent the segfault race condition/threading/timing caused.

**Pull request closed without merging**